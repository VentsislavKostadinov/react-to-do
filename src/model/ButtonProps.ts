import { BaseProps } from './BaseProps'

export interface ButtonProps extends BaseProps {
    text: string
    variant: string
    type: 'button' | 'submit'
}
