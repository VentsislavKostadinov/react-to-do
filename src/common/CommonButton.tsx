import { Button } from 'react-bootstrap'
import { ButtonProps } from '../model/ButtonProps.types'

export const CommonButton = ({
    text,
    variant,
    type,
    handleClick,
    className,
    dataTestid,
}: ButtonProps) => {
    return (
        <Button
            data-testid={dataTestid}
            className={className}
            variant={variant}
            type={type}
            onClick={handleClick}
        >
            {text}
        </Button>
    )
}
