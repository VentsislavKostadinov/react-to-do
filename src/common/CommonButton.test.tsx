import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { CommonButton } from './CommonButton'
import { ButtonProps } from '../model/ButtonProps.types'

describe('CommonButton component', () => {
    const commonButtonProps: ButtonProps = {
        dataTestid: 'add-btn',
        text: 'Add',
        variant: 'transparent',
        type: 'submit',
    }

    it('it renders correctly', () => {
        const { asFragment } = render(<CommonButton {...commonButtonProps} />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('it is visible and correct props', () => {
        
        render(
            <CommonButton {...commonButtonProps} />,
        )

        const addButton = screen.getByTestId('add-btn')
        expect(addButton).toHaveTextContent(commonButtonProps.text)
        expect(addButton).toHaveAttribute('type', 'submit')
        expect(addButton).toBeInTheDocument
    })

    it('clicking works', () => {
        const handleClick = jest.fn()
        render(
            <CommonButton handleClick={handleClick} {...commonButtonProps} />,
        )

        const addButton = screen.getByTestId('add-btn')
        fireEvent.click(addButton)
        expect(handleClick).toHaveBeenCalledTimes(1)
    })
})
