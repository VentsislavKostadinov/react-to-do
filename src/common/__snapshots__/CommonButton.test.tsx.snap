// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonButton component it renders correctly 1`] = `
<DocumentFragment>
  <button
    class="btn btn btn-transparent"
    data-testid="add-btn"
    type="submit"
  >
    Add
  </button>
</DocumentFragment>
`;
